java/memory/memory-core/src/main/java/org/apache/arrow/memory/ArrowBuf.java

```diff
@@ -1184,7 +1184,7 @@ public ArrowBuf setZero(long index, long length) {
    * @param length length of bytes to set.
    * @return this ArrowBuf
    */
-  public ArrowBuf setOne(int index, int length) {
+  public ArrowBuf setOne(long index, long length) {
```

Comment:
Could we leave behind an `@Deprecated` version of this with the original signature, then remove it in v12?